tools/driver/dump_pcm_main.cpp

```diff
@@ -0,0 +1,153 @@
+//===--- dump_pcm_main.cpp - Clang module info utility --------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Debugging aid that prints information about a precompiled Clang module
+// emitted by the Swift compiler (implicit modules, and explicit modules emitted
+// with the -emit-pcm flag).
+//
+// This is functionally equivalent to Clang's -module-file-info frontend flag,
+// but is provided here to guarantee compatibility with modules emitted by the
+// Swift compiler since those modules will not be compatible with a different
+// version of Clang (even for debug-printing).
```

Comment:
I thought Clang's `-module-file-info` was based on relatively stable parts of the Clang module file format? Personally, Could we rely on that for the "dump a module's metadata for debugging" case, rather than have this much Clang-specific logic in Swift?

Reply:
Unfortunately, the Clang used to generate the module must be the same version as the one that runs `-module-file-info` on it, or you get an error like:

```
error: PCH file built from a different branch () than the compiler ((clang-1100.0.33.8))
```

And Xcode provides no guarantee that its `clang` binary and the Clang linked into its `swiftc` binary will be the same version; over the past few versions of Swift, it's more often been the case that they're slightly different revisions. For example, Xcode 11's `clang` is `clang-1100.0.33.8` but `swiftc''s clang is `clang-1100.0.33.7`.

Reply:
Okay, thanks for correcting my faulty memory. I agree it makes sense to have this functionality in Swift.

Reply:
I've turned this into a `-dump-pcm` no-output mode in the standard driver/frontend.